index.html

```diff
@@ -208,21 +208,28 @@ <h3>
           The {{ImageResource/label}} of an {{ImageResource}} is a [=string=]
           that provides an accessible name for the associated image.
         </p>
+        <p>
+          Implementors MUST [=calculate an accessible name=] for each
+          {{ImageResource}}.
+        </p>
         <p>
           Authors are strongly encouraged to add a {{ImageResource/label}}
           unless the {{ImageResource}}'s label can be inferred from an
-          <dfn>external label</dfn> in its context. For example, the
-          [=manifest/`name`=] can act as the {{ImageResource/label}} for any
-          member of the [=manifest/`icons`=] array.
+          <dfn>external label</dfn> in its context. For example, the context of
+          the [=manifest=], the [=manifest/`name`=] (or
+          [=manifest/`short_name`=]) provides an [=external label=] for all members of
+          the [=manifest/`icons`=] array that don’t have an explicit
+          {{ImageResource/label}} defined.
         </p>
         <p>
-          If an explicit {{ImageResource/label}} is not provided, but the
-          {{ImageResource/type}} supports providing an [=embedded accessible
-          name=], implementors MAY choose to infer the {{ImageResource/label}}
-          until the {{ImageResource/src}} has been downloaded and parsed. If
-          the {{ImageResource/src}} is found to provide an [=embedded
-          accessible name=], implementors may choose to update the
-          {{ImageResource/label}} to match that value.
+          If neither an explicit {{ImageResource/label}} nor an [=external
+          label=] is provided, but the {{ImageResource/type}} supports
+          providing an [=embedded accessible name=], implementors MAY choose to
+          defer assigning the {{ImageResource/label}} until the
+          {{ImageResource/src}} has been downloaded and parsed. If the
+          {{ImageResource/src}} is found to provide an [=embedded accessible
+          name=], implementors MUST update the {{ImageResource/label}} to match
+          that value.
```

Comment:
This is done in the algorithm, so no need to state it here. 

```suggestion
          {{ImageResource/src}} has been downloaded and parsed. 
```